### PR TITLE
IBP-5031: Implement samples search and refactor samples service

### PR DESCRIPTION
**Pull request by cheryleigh**

### CODE REVIEW

**Review by mderamos — CHANGES_REQUESTED** [review]

File: src/main/java/org/ibp/api/brapi/v2/SampleResourceBrapi.java [hunk 1]
```diff
@@ -5,79 +5,173 @@
...
+
+	@ApiOperation(value = "Get samples", notes = "Get samples")
+	@RequestMapping(value = "/{crop}/brapi/v2/samples", method = RequestMethod.GET)
+	@JsonView(BrapiView.BrapiV2.class)
+	@ResponseBody
+	public ResponseEntity<EntityListResponse<SampleObservationDto>> getSamples(@PathVariable final String crop,
+		@ApiParam(value = "the internal DB id for a sample")
+		@RequestParam(value = "sampleDbId", required = false) final String sampleDbId,
+		@ApiParam(value = "the internal DB id for an observation unit where a sample was taken from")
+		@RequestParam(value = "observationUnitDbId", required = false) final String observationUnitDbIds,
```
Comment (mderamos) [inline src/main/java/org/ibp/api/brapi/v2/SampleResourceBrapi.java:56]: change variable name to observationUnitDbId

**Review by mderamos — CHANGES_REQUESTED** [review]

**Review by mderamos — APPROVED** [review]

File: src/main/java/org/ibp/api/brapi/v2/SampleResourceBrapi.java [hunk 2]
```diff
@@ -5,79 +5,173 @@
...
+					return SampleResourceBrapi.this.sampleServiceBrapi.countSampleObservations(requestDTO);
+				}
+
+				@Override
+				public List<SampleObservationDto> getResults(final PagedResult<SampleObservationDto> pagedResult) {
+					return SampleResourceBrapi.this.sampleServiceBrapi.getSampleObservations(requestDTO, pageRequest);
+				}
+			});
+
+		final Result<SampleObservationDto> results = new Result<SampleObservationDto>().withData(resultPage.getPageResults());
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/brapi/v2/SampleResourceBrapi.java:92]: Extract to common method:
`final Result<SampleObservationDto> results = new Result<SampleObservationDto>().withData(resultPage.getPageResults());
		final Pagination pagination = new Pagination().withPageNumber(resultPage.getPageNumber()).withPageSize(resultPage.getPageSize())
			.withTotalCount(resultPage.getTotalResults()).withTotalPages(resultPage.getTotalPages());

		final Metadata metadata = new Metadata().withPagination(pagination);
		return new ResponseEntity<>(new EntityListResponse<>(metadata, results), HttpStatus.OK);`

File: src/main/java/org/ibp/api/java/impl/middleware/sample/SampleService.java [hunk 3]
```diff
@@ -17,8 +17,4 @@
 	long countFilter(final String obsUnitId, final Integer listId);
 
 	List<SampleDTO> getGermplasmSamples(Integer gid);
-
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/java/impl/middleware/sample/SampleService.java:20]: Remove unused imports

File: src/main/java/org/ibp/api/java/impl/middleware/sample/SampleServiceImpl.java [hunk 4]
```diff
@@ -41,14 +41,4 @@ public long countFilter(final String obsUnitId, final Integer listId) {
 		this.germplasmValidator.validateGids(errors, Collections.singletonList(gid));
 		return this.sampleService.getByGid(gid);
 	}
-
-	@Override
-	public List<SampleObservationDto> getSampleObservations(final SampleSearchRequestDTO requestDTO, final Pageable pageable) {
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/java/impl/middleware/sample/SampleServiceImpl.java:46]: Remove unused imports

File: src/main/java/org/ibp/api/brapi/v2/SampleResourceBrapi.java [hunk 5]
```diff
@@ -5,79 +5,173 @@
 import io.swagger.annotations.ApiOperation;
 import io.swagger.annotations.ApiParam;
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/brapi/v2/SampleResourceBrapi.java:6]: Move this under org.ibp.api.brapi.v2.sample package

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**